### PR TITLE
Convert api helpers to typescipt

### PR DESCRIPTION
**Pull request by Pespiri**

### CODE REVIEW

File: src/api/secrets.ts [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
+  secretName: string,
+  value: string
+): Promise<string> {
   const encAppName = encodeURIComponent(appName);
   const encEnvName = encodeURIComponent(envName);
   const encComponentName = encodeURIComponent(componentName);
   const encSecretName = encodeURIComponent(secretName);
 
-  const body = { secretValue: value.toString() };
+  const body: { secretValue: string; type?: SecretType } = {
```
Comment (nilsgstrabo) [inline src/api/secrets.ts:18]: Why is type defined when it is not used?

File: src/models/model-types.ts [hunk 2]
```diff
@@ -3,8 +3,18 @@ type TestDescriptorType = {
...
 
+type ToString<T> = T extends Date
+  ? string
+  : T extends object
+  ? RawModel<T>
+  : string;
+
 export type TestDependencyDataType<T = unknown> = Array<TestDescriptorType & T>;
 
+export type RawModel<T> = {
```
Comment (nilsgstrabo) [inline src/models/model-types.ts:14]: Can you explain what RawModel is?

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
Added a comment and a question

File: src/api/secrets.ts (hunk 1 above)
Comment (Pespiri) [reply src/api/secrets.ts:18]: It's a valid field for this request for the API, so I just added it

File: src/models/model-types.ts (hunk 2 above)
Comment (Pespiri) [reply src/models/model-types.ts:14]: RawModel recursively types any property of a model into string properties, which can be useful for marking an unprocessed raw data model from the API that has not been normalized, I just couldn't figure a better generic name for it

```ts
interface Model {
  isSet: boolean;
  name: string;
  count: number;
  object: {
    comment: string;
    date: Date;
  };
}

// marking an object with RawModel results in all properties being typed into string
// i.e variable.count, variable.object.date, etc. are now typed as string
let variable: RawModel<Model>;
```

File: src/api/secrets.ts (hunk 1 above)
Comment (nilsgstrabo) [reply src/api/secrets.ts:18]: I checked radix-api, and the field is declared, but never used. It was added by Sergey 11 months ago. We'll ask him when he returns from vacation. I think we can safely remove the type field from secrets.ts

File: src/models/model-types.ts (hunk 2 above)
Comment (nilsgstrabo) [reply src/models/model-types.ts:14]: And why do we need to do this?

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**